src/arena.c

```diff
@@ -491,6 +491,20 @@ arena_extent_ralloc_large_expand(tsdn_t *tsdn, arena_t *arena, extent_t *extent,
 	arena_nactive_add(arena, udiff >> LG_PAGE);
 }
 
+static ssize_t
+arena_decay_time_read(arena_t *arena) {
+	arena_decay_time_t dt;
+	dt.u = atomic_read_zu(&arena->decay.time.u);
+	return dt.s;
```

Comment:
This must be something that's diverged in C and C++, and been clarified in C. I have had experience in the past with C compilers optimizing away code like this.  But looking at http://en.cppreference.com/w/c/language/union now says:

"If the member used to access the contents of a union is not the same as the member last used to store a value, the object representation of the value that was stored is reinterpreted as an object representation of the new type (this is known as type punning)."

So this is fine. Thanks!